iocore/net/SSLCertLookup.cc

```diff
@@ -471,11 +492,12 @@ SSLContextStorage::lookup(const char *name) const
   }
 
   // Then strip off the top domain name and look for a wildcard domain match
-  const char *subdomain = index(name, '.');
+  const char *subdomain = index(lower_case_name, '.');
   if (subdomain) {
     ++subdomain; // Move beyond the '.'
-    if (ink_hash_table_lookup(const_cast<InkHashTable *>(this->wilddomains), subdomain, &value)) {
-      return &(this->ctx_store[reinterpret_cast<intptr_t>(value)]);
+    auto index = this->wilddomains.get(subdomain);
+    if (index >= 0) {
```

Comment:
It seems that HashMap<K, AHashFns, C, A>::get(K akey) returns 0 if akey is not in the hash table, so index >= 0 doesn't feel right.